packages/ui5-application-inquirer/src/prompts/prompt-helpers.ts

```diff
@@ -0,0 +1,226 @@
+import type { PromptSeverityMessage, YUIQuestion, validate } from '@sap-ux/inquirer-common';
+import { existsSync } from 'fs';
+import type { Answers, Question } from 'inquirer';
+import { join } from 'path';
+import { coerce, gte } from 'semver';
+import { defaultProjectNumber, t } from '../i18n';
+import {
+    promptNames,
+    type CommonPromptOptions,
+    type PromptDefaultValue,
+    type UI5ApplicationAnswers,
+    type UI5ApplicationPromptOptions,
+    type UI5ApplicationQuestion
+} from '../types';
+import { latestVersionString } from '@sap-ux/ui5-info';
+
+/**
+ * Tests if a directory with the specified `appName` exists at the path specified by `targetPath`.
+ *
+ * @param appName directory name of application
+ * @param targetPath directory path where application directory would be created
+ * @returns true, if the combined path exists otherwise false
+ */
+export function appPathExists(appName: string, targetPath?: string): boolean | string {
+    return existsSync(join(targetPath ?? process.cwd(), appName.trim()));
+}
+/**
+ * Generate a default applicaiton name that does not exist at the specified path.
+ *
+ * @param targetPath the target path where the application directory would be created
+ * @returns a suggested application name that can be created at the specified target path
+ */
+export function defaultAppName(targetPath: string): string {
+    let defProjNum = defaultProjectNumber;
+    let defaultName = t('prompts.appNameDefault');
+    while (exports.appPathExists(`${defaultName}`, targetPath)) {
+        defaultName = t('prompts.appNameDefault', { defaultProjectNumber: ++defProjNum });
+        // Dont loop forever, user will need to provide input otherwise
+        if (defProjNum > 999) {
+            break;
+        }
+    }
+    return defaultName;
+}
+
+/**
+ * Checks if the specified semantic version string is greater than or equal to the minimum version.
+ * If the specified version is not a parseable semantic version, returns true.
+ *
+ * @param version the version to test
+ * @param minVersion the minimum version to test against
+ * @returns - true if the specified version is greater than or equal to the minimum version, or the version is not a coercable semver
+ */
+export function isVersionIncluded(version: string, minVersion: string): boolean {
+    // Extract a usable version, `snapshot`, `latest` etc will be ignored
+    const ui5SemVer = coerce(version);
+    if (ui5SemVer) {
+        return gte(ui5SemVer, minVersion);
+    }
+    return version === latestVersionString;
+}
+
+/**
+ * Adds additional conditions to the provided questions.
+ *
+ * @param questions the questions to which the condition will be added
+ * @param condition function which returns true or false
+ * @returns the passed questions reference
+ */
+export function withCondition(questions: Question[], condition: (answers: Answers) => boolean): Question[] {
```

Comment:
Would this be suitable in the inquirer common module? 
Or we will end up with another common module,  I presume it will be useful for service inquirer. 
Same goes for `extendAdditionalMessages` etc

Reply:
Id prefer not to move anything to a common module until necessary. It may be used later and should be moved at thst stage. Thx